pkg/operators/v1alpha1/installplan_types.go

```diff
@@ -227,6 +228,7 @@ func ConditionMet(cond InstallPlanConditionType, now *metav1.Time) InstallPlanCo
 type Step struct {
 	Resolving string       `json:"resolving"`
 	Resource  StepResource `json:"resource"`
+	Optional  bool         `json:"optional,omitempty"`
```

Comment:
Quick question: I haven't read through the original enhancement entirely so apologies if has already been addressed - does this need to be a pointer to a bool?

Reply:
@timflannagan Thanks for looking at it. It is indeed the way it was specified in the[ enhancement proposal](https://github.com/operator-framework/enhancements/blob/master/enhancements/optional-manifest.md#step). That said it is possible to amend it if there is a case for it.
I understand that a bool is not extensible (it accepts two values and that's it). Originally I had played with the idea of having a field named Criticality instead of Optional, accepting values like: "Mandatory", "Optional" and possibly others. The thing is that I was advised to have the list of optional manifests in a bundle property rather than having the criticality specified in an annotation directly in the manifest. As an entry in a list the file is either there or not. This is binary... like a boolean. Keeping the same interface I don't see any value in making it a string but I am open to read your views.

Reply:
Yeah definitely - IIRC, the need for a pointer to a bool in relevant when we need to distinguish between a user supplied value (i.e. it was explicitly set) vs. the default/empty value from Go in the context of boolean types. If that isn't the case here, then we should be all set to settle on a bool.

Reply:
ok. Thanks for clarifying I had missed the pointer aspect.